thirdparty/github.go

```diff
@@ -383,6 +383,35 @@ func GetCommitDiff(ctx context.Context, oauthToken, repoOwner, repo, sha string)
 	return commit, nil
 }
 
+// GetBranchCommitHash retrieves the most recent commit hash for branch for the given repo, module, and branch name.
+func GetBranchCommitHash(ctx context.Context, repo, moduleBranch, token string) (string, error) {
```

Comment:
I'm slightly confused why we needed to move this; if this is only used in the units package should it just be a private function there?

Reply:
Moved back to units